modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/ZonePartitionReplicaListener.java

```diff
@@ -17,95 +17,209 @@
 import org.apache.ignite.internal.replicator.listener.ReplicaListener;
+import org.apache.ignite.internal.replicator.message.ReadOnlyDirectReplicaRequest;
 import org.apache.ignite.internal.replicator.message.ReplicaRequest;
+import org.apache.ignite.internal.replicator.message.SchemaVersionAwareReplicaRequest;
 import org.apache.ignite.internal.replicator.message.TableAware;
+import org.apache.ignite.internal.tx.TransactionIds;
 import org.apache.ignite.internal.tx.message.TxFinishReplicaRequest;
 import org.apache.ignite.internal.tx.message.TxMessagesFactory;
+import org.jetbrains.annotations.Nullable;
 import org.jetbrains.annotations.VisibleForTesting;
 
 /**
  * Zone partition replica listener.
  */
 public class ZonePartitionReplicaListener implements ReplicaListener {
+    private static final IgniteLogger LOG = Loggers.forClass(ZonePartitionReplicaListener.class);
+
     private static final TxMessagesFactory TX_MESSAGES_FACTORY = new TxMessagesFactory();
 
-    private static final IgniteLogger LOG = Loggers.forClass(ZonePartitionReplicaListener.class);
+    /** Factory to create RAFT command messages. */
+    private static final PartitionReplicationMessagesFactory PARTITION_REPLICATION_MESSAGES_FACTORY =
+            new PartitionReplicationMessagesFactory();
 
     // TODO: https://issues.apache.org/jira/browse/IGNITE-22624 await for the table replica listener if needed.
     private final Map<TablePartitionId, ReplicaListener> replicas = new ConcurrentHashMap<>();
 
+    /** Raft client. */
     private final RaftCommandRunner raftClient;
 
+    /** Clock service. */
+    private final ClockService clockService;
+
+    // TODO Probably the list of handlers should be moved to a map or something appropriate.
+    // Handler for minimum active tx time request.
+    private final MinimumActiveTxTimeReplicaRequestHandler minimumActiveTxTimeReplicaRequestHandler;
+
+    /** Zone replication group id. */
+    ZonePartitionId replicationGroupId;
+
     /**
      * The constructor.
      *
+     * @param replicationGroupId Zone replication group identifier.
+     * @param clockService Clock service.
      * @param raftClient Raft client.
      */
-    public ZonePartitionReplicaListener(RaftCommandRunner raftClient) {
+    public ZonePartitionReplicaListener(
+            ZonePartitionId replicationGroupId,
+            ClockService clockService,
+            RaftCommandRunner raftClient) {
+        this.replicationGroupId = replicationGroupId;
+        this.clockService = clockService;
         this.raftClient = raftClient;
+
+        // Request handlers initialization.
+        minimumActiveTxTimeReplicaRequestHandler = new MinimumActiveTxTimeReplicaRequestHandler(
+                PARTITION_REPLICATION_MESSAGES_FACTORY,
+                clockService,
+                this::applyCmdWithExceptionHandling
+        );
     }
 
     @Override
     public CompletableFuture<ReplicaResult> invoke(ReplicaRequest request, UUID senderId) {
```

Comment:
I've expected that you will copy 
```
        return ensureReplicaIsPrimary(request)
                .thenCompose(res -> processRequest(request, res.get1(), senderId, res.get2()))
                .thenApply(res -> {
                    if (res instanceof ReplicaResult) {
                        return (ReplicaResult) res;
                    } else {
                        return new ReplicaResult(res, null);
                    }
                });
```
With empty ensureReplicaIsPrimary.

Anyway let's wait for https://github.com/apache/ignite-3/pull/5187. There are too many clashes for now.